filecoin-proofs/src/api/mod.rs

```diff
@@ -530,25 +517,31 @@ unsafe fn try_into_post_proofs_bytes(
     flattened_proofs_ptr: *const u8,
     flattened_proofs_len: libc::size_t,
 ) -> crate::error::Result<Vec<Vec<u8>>> {
-    post_proof_partitions::try_from_u8(proof_partitions)
-        .map(PoStProofBytesAmount::from)
-        .map(usize::from)
-        .map(|chunk| into_proof_vecs(chunk, flattened_proofs_ptr, flattened_proofs_len))
+    let chunk_size = proof_partitions as usize * SINGLE_PARTITION_PROOF_LEN;
+
+    if flattened_proofs_len % chunk_size != 0 {
+        Err(format_err!(
+            "proofs array len={:?} incompatible with partitions={:?}",
+            flattened_proofs_len,
+            proof_partitions
+        ))
```

Comment:
tip: I believe you can use [ensure](https://docs.rs/failure/0.1.5/failure/macro.ensure.html) to write this a bit more compact:

```rust
ensure!(a != 0, "error message... {} - {}", a, b);
```

Reply:
I like! Did not know.